stdpopsim/dfe.py

```diff
@@ -196,27 +202,32 @@ class DFE:
     mutation_types = attr.ib(default=attr.Factory(list))
     proportions = attr.ib(default=attr.Factory(list))
     citations = attr.ib(default=attr.Factory(list))
+    qc_dfe = attr.ib(default=None)
 
     def __attrs_post_init__(self):
         self.citations = [] if self.citations is None else self.citations
         if self.proportions == [] and len(self.mutation_types) == 1:
             self.proportions = [1]
 
-        if not (
-            isinstance(self.proportions, collections.abc.Collection)
-            and isinstance(self.mutation_types, collections.abc.Collection)
-            and len(self.proportions) == len(self.mutation_types)
-        ):
+        if not (isinstance(self.proportions, list)):
+            raise ValueError("proportions must be a list.")
+
+        if not (isinstance(self.mutation_types, list)):
+            raise ValueError("mutation_types must be a list.")
+
+        if not (len(self.proportions) == len(self.mutation_types)):
             raise ValueError(
                 "proportions and mutation_types must be lists of the same length."
             )
 
         for p in self.proportions:
             if not isinstance(p, (float, int)) or p < 0:
                 raise ValueError("proportions must be nonnegative numbers.")
-        sum_p = sum(self.proportions)
-        if not np.isclose(sum_p, 1):
-            raise ValueError("proportions must sum 1.0.")
+
+        if (len(self.proportions) > 0) and (len(self.mutation_types) > 0):
```

Comment:
We don't need to check the length of `mutation_types` here (we already know it's the same length as proporitons).
```suggestion
        if len(self.proportions) > 0):
```